### ISSUE TITLE
Issues of OpenMathInstruct-2 dataset decontamination

### ISSUE THREAD

**Issue by henryj18** [Kipok/NeMo-Skills#235]
The paper claims that decontamination is performed, but if you check the dataset on hugginaface, there are problems with source as "MATH", "GSM8K" etc. These two are very well-known benchmarks to evaluate performance, and the OpenMathInstruct-2 paper does show some perforamcen measurement of training Llama 3.1 on OpenMathInstruct-2. How could the performance be evaluated against a dataset that is part of the training dataset, i.e. the "MATH" dataset? Furthermore, according to the description of "decontamination" in the paper, problems in "MATH" dataset should have been removed from  OpenMathInstruct-2, but they are still in it.

**Comment by Kipok** [comment]
MATH and GSM8K have both training and test subsets. We decontaminate against the test part (which is what is used to calculate final accuracy) and include training subsets of both datasets in our training data (as well as use them as seed for problem augmentation)
